iothub/device/src/Transport/Http/HttpClientHelper.cs

```diff
@@ -237,5 +237,10 @@ private static async Task<T> ReadAsAsync<T>(HttpContent content, CancellationTok
             using var jsonReader = new JsonTextReader(reader);
             return new JsonSerializer().Deserialize<T>(jsonReader);
         }
+
+        public void Dispose()
+        {
+            throw new NotImplementedException();
```

Comment:
The reason this class is flagged with CA1001 is because it creates a disposable object (HttpClient) and needs to dispose it. The expected implementation here is to call dispose on all the disposable instances this class owns, like:

```csharp
_httpClientObj?.Dispose();
```

Reply:
Generally, ```NotImplementedException``` is reserved for cases where a developer hasn't written the body of the method yet but wants to be able to build/run tests that don't use this method.

We really wouldn't ever check in something like that since our codebase isn't that early on in its development.

Reply:
Makes sense, fixed